### PR TITLE
Rename dataset.* to datastream.*

### PR DESCRIPTION
**Pull request by mtojek**
This PR adjusts the validation to check datastream fields instead of dataset.

### CODE REVIEW

File: testdata/package/datasources/1.0.0/dataset/examplelog1/fields/base-fields.yml [hunk 1]
```diff
@@ -1,12 +1,12 @@
-- name: dataset.type
+- name: datastream.type
   type: constant_keyword
   description: >
     Dataset type.
-- name: dataset.name
+- name: datastream.name
```
Comment (ruflin) [inline testdata/package/datasources/1.0.0/dataset/examplelog1/fields/base-fields.yml:5]: It is `datastream.dataset` now.

File: testdata/package/datasources/1.0.0/dataset/examplelog2/fields/base-fields.yml [hunk 2]
```diff
@@ -1,12 +1,12 @@
-- name: dataset.type
+- name: datastream.type
   type: constant_keyword
   description: >
     Dataset type.
-- name: dataset.name
+- name: datastream.name
```
Comment (ruflin) [inline testdata/package/datasources/1.0.0/dataset/examplelog2/fields/base-fields.yml:5]: See above.

File: testdata/package/datasources/1.0.0/dataset/examplelog1/fields/base-fields.yml (hunk 1 above)
Comment (mtojek) [reply testdata/package/datasources/1.0.0/dataset/examplelog1/fields/base-fields.yml]: Fixed

File: testdata/package/datasources/1.0.0/dataset/examplelog2/fields/base-fields.yml (hunk 2 above)
Comment (mtojek) [reply testdata/package/datasources/1.0.0/dataset/examplelog2/fields/base-fields.yml]: Fixed

**Pull request merged**